autogen/agentchat/conversable_agent.py

```diff
@@ -601,12 +603,16 @@ def generate_oai_reply(
             return False, None
         if messages is None:
             messages = self._oai_messages[sender]
-
-        # TODO: #1143 handle token limit exceeded error
-        response = oai.ChatCompletion.create(
-            context=messages[-1].pop("context", None), messages=self._oai_system_message + messages, **llm_config
-        )
-        return True, oai.ChatCompletion.extract_text_or_function_call(response)[0]
+        while True:
+            try:
+                response = oai.ChatCompletion.create(
+                    context=messages[-1].pop("context", None), messages=self._oai_system_message + messages, **llm_config
+                )
+                return True, oai.ChatCompletion.extract_text_or_function_call(response)[0]
+            except openai.error.InvalidRequestError as e:
```

Comment:
The `InvalidRequestError` doesn't always mean context window overflow. Other type of `InvalidRequestError` shouldn't be handled by this approach.
Also, we should use this method only when users explicitly ask to. I think we can replace line 615 with a more generic function `compress_messages` which is configurable by users.
@kevin666aa what do you think?

Reply:
I probably should have looked at other PRs first. Is this made redundant by https://github.com/microsoft/autogen/pull/131 ? Or maybe still a valid stop-gap?

Reply:
In #131 , here is the usage of compression:
Before calling openai, a `on_oai_token_limit` is registered
- If compress_config set to false, the function would check if the existing token exceeds max token limit of a model. If so, it will terminate the conversation early to avoid token_limit_error.
- If compress_config set to true, the function would compress previous messages if user's preset token threshold is reached.

Reply:
It is not clear to me if any action should be taken on my end here. I intend to defer to the preference of the repo maintainers. Please let me know if there is anything specific I can contribute here. Thanks for the engagement so far y'all.

Reply:
@kevin666aa is your PR compatible with this PR? How soon will you finish it? We have two paths now: merge this PR first (after necessary modification) and then base your PR off it; or merge your PR first and then base this PR of it. I generally prefer smaller, faster PRs than a giant and slow PR. Which PR requires lower effort to be ready to merge?

Reply:
This PR is compatible with mine. My PR handles messages between the request, and this PR handles it after the error. I think we can merge this if it is quicker.

Reply:
Thanks. Could you suggest minimal changes required to merge this PR?

Reply:
@kevin666aa

Reply:
@kevin666aa what's your opinion about my comment:

The InvalidRequestError doesn't always mean context window overflow. Other type of InvalidRequestError shouldn't be handled by this approach.
Also, we should use this method only when users explicitly ask to. I think we can replace line 615 with a more generic function compress_messages which is configurable by users.

Reply:
I agree that `InvalidRequestError` is not equal to context window overflow, but I haven't see other types of  InvalidRequestError so far. This can be added as an option in Compression agent. I have changed it so that use can pass in different mode, current there are "TERMINATE" which terminates the chat, and "COMPRESS" . We can add a "DELETE" in it.

Reply:
> @kevin666aa what's your opinion about my comment:
> 
> The InvalidRequestError doesn't always mean context window overflow. Other type of InvalidRequestError shouldn't be handled by this approach. Also, we should use this method only when users explicitly ask to. I think we can replace line 615 with a more generic function compress_messages which is configurable by users.

I agree that `InvalidRequestError` is not equal to context window overflow, but I haven't see other types of  InvalidRequestError so far. This can be added as an option in Compression agent. I have changed it so that use can pass in different mode, current there are "TERMINATE" which terminates the chat, and "COMPRESS" . We can add a "DELETE" in it.

Reply:
I have seen other types of InvalidRequestError. For example, when input arguments violate requirements.

Reply:
Is there a way to differentiate based on the exception? Can we assume the message dialog won't exceed five by the time this error occurs? Functionally it does not seem to matter as my loop will terminate after enough messages are removed.

Reply:
Then it is better to add it as an option in compression agent. @bioshazard Can you take a look at #131 if you have time? It would be great if you can give it a test.

Also, from what I have now, you can add this PR in it quickly by add a mode "DELETE" in `compress_config`, and then implement the corresponding method in function `on_oai_token_limit`